test/parallel/test-net-reconnect-error.js

```diff
@@ -10,8 +10,7 @@ let disconnect_count = 0;
 const c = net.createConnection(common.PORT);
 
 c.on('connect', function() {
-  console.error('CLIENT connected');
-  assert.ok(false);
+  common.fail('client should not have connected');
```

Comment:
Yeah, but there are other cases where this is already done, for example https://github.com/nodejs/node/pull/10899/files#diff-74c7cfcd74dc6766c9877fac9daf0721R19.

I'm also not a fan of using `common.fail()` when the argument is not a string for example when it used as `'request' `listener, `'error'` listener, etc.